pypeit/core/coadd.py

```diff
@@ -1294,67 +1294,234 @@ def order_phot_scale(spectra, phot_scale_dicts, nsig=3.0, niter=5, debug=False):
-                plt.plot(spectra.data['wave'][iord - 1, allok_iordm1], spectra.data['flux'][iord - 1, allok_iordm1], 'b-',
-                         label='scaled spectrum')
-                mny, medy, stdy = stats.sigma_clipped_stats(fluxes[iord, allok], sigma=nsig, iters=niter)
+                plt.plot(wave[allok_iord_ref], fluxes_out[iord_ref, allok_iord_ref], '-',lw=10,color='0.7', label='Scale region')
+                plt.plot(wave[allok_iord_scl], fluxes_out[iord,allok_iord_scl], 'r-', label='reference spectrum')
+                plt.plot(wave[allok_iord_scl], fluxes_in[iord,allok_iord_scl], 'k-', label='raw spectrum')
+                plt.plot(wave[allok_iord_scl], fluxes_out[iord, allok_iord_scl], 'b-',label='scaled spectrum')
+                mny, medy, stdy = stats.sigma_clipped_stats(fluxes_in[iord, allok_iord_scl], sigma=nsig, iters=niter)
                 plt.ylim([0.1 * medy, 4.0 * medy])
-                plt.xlim([np.min(wave[sigs[iord - 1, :] > 0]), np.max(wave[sigs[iord, :] > 0])])
+                plt.xlim([np.min(wave[allok_iord_ref]), np.max(wave[allok_iord_scl])])
                 plt.legend()
                 plt.xlabel('wavelength')
                 plt.ylabel('Flux')
                 plt.show()
         else:
-            msgs.warn('Not enough overlap region for sticking different orders.')
+            msgs.warn('Not enough spectral overlap to rescale spectra in order {:d}.'.format(iord) + ' Not recaling for this order'
+                      'Consider decreasing min_overlap_frac = {:5.3f}'.format(min_overlap_frac))
+            fluxes_out = fluxes_in[iord-1,:]
+            sigs_out = sigs_in[iord-1,:]
+
+
+
+def merge_order(spectra, wave_grid, extract='OPT', orderscale='median', niter=5, sigrej_final=3., SN_MIN_MEDSCALE = 5.0,
+                overlapfrac = 0.01, num_min_pixels=10,phot_scale_dicts=None, qafile=None, outfile=None, debug=False):
+    """
+        routines for merging orders of echelle spectra.
+    parameters:
+        spectra: spectra in the format of Xspectrum1D
+        wave_grid (numpy array): The final wavelength grid for the output. Note that the wave_grid should be
+          consistent with your spectra, if not please rebin it first.
+        extract (str): 'OPT' or 'BOX'
+        orderscale (str): which method you want use to scale different orders: median, photometry or None.
+        niter (int): number of iteration for rejections in the overlapped part
+        sigrej_final (float): sigma for rejections in the overlapped part
+        SN_MIN_MEDSCALE (float): minimum SNR for scaling different orders
+        overlapfrac (float): minimum overlap fraction for scaling different orders.
+        phot_scale_dicts (dict): A dict contains photometric information of each order. An example is given below.
+          phot_scale_dicts = {'0': {'filter': None, 'mag': None, 'mag_type': None, 'masks': None},
```

Comment:
Add two colons so that sphinx will interpret this as a code block:
```
        phot_scale_dicts (dict):
            Photometric information for each order.  Format is::

                phot_scale_dicts = {'0': {'filter': None, 'mag': None, 'mag_type': None, 'masks': None},
                ...
```